src/lib/operators/delete.cpp

```diff
@@ -70,7 +70,11 @@ void Delete::_finish_commit() {
 
   const auto table_statistics = _table->table_statistics();
   if (table_statistics) {
-    table_statistics->increment_invalid_row_count(num_rows_deleted);
+    // Hackyhacky const cast. We're not changing the table, technically, just its statistics.
+    // This updates the TableStatistics row count by creating and setting new TableStatistics in the Table.
+    std::const_pointer_cast<Table>(_table)->set_table_statistics(std::make_shared<TableStatistics>(
```

Comment:
No need for const cast since `_table` is not const.
Definition in `delete.hpp`: `std::shared_ptr<Table> _table;`

Reply:
@FabianWiebe: Is this the result of a thorough review or just something you found when you looked at the PR? Both is fine, it's just that when I see "FabianWiebe reviewed a day ago", I wonder if you have seen the rest or not.

A short comment on the review if you've only looked at parts would be great.

Reply:
This is just a comment for this section. Not a full PR review. I did not go in detail through the rest.

Reply:
fixed